From 09b6373593b5bb52629e2001687997d2b29be4f6 Mon Sep 17 00:00:00 2001
From: Wen Chen <chenwen54@huawei.com>
Date: Mon, 27 Apr 2026 03:31:22 +0800
Subject: [PATCH] urma: Enable ctp while using single_path mode and fix the bug
 that urma_user_ctl returns success while bonding device is in use.

---
 src/urma/lib/urma/bond/bondp_provider_ops.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/urma/lib/urma/bond/bondp_provider_ops.c b/src/urma/lib/urma/bond/bondp_provider_ops.c
index d39aeee..8d4e09b 100644
--- a/src/urma/lib/urma/bond/bondp_provider_ops.c
+++ b/src/urma/lib/urma/bond/bondp_provider_ops.c
@@ -506,6 +506,13 @@ int bondp_set_bonding_mode(urma_context_t *ctx, bondp_bonding_mode_t bonding_mod
         return -EINVAL;
     }
 
+    uint64_t cnt = (uint64_t)atomic_load(&ctx->ref.atomic_cnt);
+    if (cnt > 1) {
+        URMA_LOG_WARN("already in use, atomic_cnt: %lu, dev_name: %s.\n",
+            cnt, ctx->dev->name);
+        return URMA_EAGAIN;
+    }
+
     if (bonding_mode < 0 || bonding_mode >= BONDP_BONDING_MODE_MAX) {
         URMA_LOG_ERR("Invalid bonding mode: %d\n", bonding_mode);
         return -EINVAL;
-- 
2.28.0.windows.1

